types/connect/index.d.ts

```diff
@@ -1,94 +1,163 @@
+            /** The Node.js http request object. */ req: Request,
+            /** The Node.js http response object. */ res: http.ServerResponse,
+            /** A function to call to invoke the next middleware. */ next: NextFunction
+        ) => void): Server;
+
+        /** Register a middleware. */
         use(fn: HandleFunction): Server;
+        /** Mount a middleware on the specified route. */
         use(route: string, fn: HandleFunction): Server;
 
         /**
-        * Handle server requests, punting them down
-        * the middleware stack.
-        *
-        * @private
-        */
-        handle(req: http.IncomingMessage, res: http.ServerResponse, next: Function): void;
+         * Calling the function will run the middleware stack against the given Node.js http request (`req`)
+         * and response (`res`) objects. An optional function `out` can be provided that will be called if the
+         * request (or error) was not handled by the middleware stack.
+         */
+        handle(req: http.IncomingMessage, res: http.ServerResponse, out?: NextFunction): void;
 
         /**
-        * Listen for connections.
-        *
-        * This method takes the same arguments
-        * as node's `http.Server#listen()`.
-        *
-        * HTTP and HTTPS:
-        *
-        * If you run your application both as HTTP
-        * and HTTPS you may wrap them individually,
-        * since your Connect "server" is really just
-        * a JavaScript `Function`.
-        *
-        *      var connect = require('connect')
-        *        , http = require('http')
-        *        , https = require('https');
-        *
-        *      var app = connect();
-        *
-        *      http.createServer(app).listen(80);
-        *      https.createServer(options, app).listen(443);
-        *
-        * @api public
-        */
-        listen(port: number, hostname?: string, backlog?: number, callback?: Function): http.Server;
-        listen(port: number, hostname?: string, callback?: Function): http.Server;
-        listen(path: string, callback?: Function): http.Server;
-        listen(handle: any, listeningListener?: Function): http.Server;
+         * Start the app listening for requests. This method will internally create a Node.js HTTP server and
+         * call `.listen()` on it.
+         *
+         * This is an alias to the `server.listen()` method in the version of Node.js running, so consult
+         * the Node.js documentation for all the different variations. The most common signature is `app.listen(port)`.
+         */
+        listen(port: number, hostname?: string, backlog?: number, listeningListener?: () => void): http.Server;
+        listen(port: number, hostname?: string, listeningListener?: () => void): http.Server;
+        listen(path: string, listeningListener?: () => void): http.Server;
+        listen(handle: any, listeningListener?: () => void): http.Server;
```

Comment:
using `listeningListener?: () => void` instead of Function, to match signature of Server.listen

Reply:
I could have also added the other overloads of Server.listen (since it's the exact same method) but I wasn't sure if I should do that (I only looked at the latest version of Node) so I left only the ones already present.